### ISSUE TITLE
Add "PKI as a Token" to Assertion Examples

### ISSUE THREAD

**Issue by twbaldridge** [usnistgov/800-63-3#587]
### All Fields Are Required
**Organization Name (N/A, if individual)**: DoDMDC

**Organization Type (see below for codes)**: 1
 
**Document (63-3, 63A, 63B, or 63C)**: 63C

**Reference (Include section and paragraph number)**: 11. Assertion Examples
 
**Comment (Include rationale for comment)**:
 
Add a forth assertion technology based on Public Key Infrastructure (PKI) as a token.
 
**Suggested Change**:

Add a forth Section that describes the mechanism for an in-line outbound TLS terminate and regenerate connection capability.  In this framework the outbound perimeter device verifies the users authenticator, up to and including AAL 3, obtains IAL information from the users validated PKI certificate and then requests a short validity period PKI certificate using a private key controlled by the perimeter device to represent the user to the relying party requested by the user.

 
---
 
Organization Type: 1 = Federal, 2 = Industry, 3 = Academia, 4 = Self, 5 = Other

**Comment by twbaldridge** [comment]
**Dynamically Generated PKI Certificates as a Token Substitute**

In a network environment where an internal PKI-credentialed network user is accessing an external resource outside the network, it can be difficult for that user to authenticate to the resource with a high level of assurance when there are intermediate connections between the two. To be specific, boundary devices that intercept traffic for inspection disrupt the mutual authentication / two-way TLS connection the user needs to assuredly authenticate to the external resource.

The most secure solution is for the boundary device, in this case, to authenticate the user and generate an authentication assertion about the user, often called an authentication token, that is passed to the resource. However, many resources do not have the software to accept such tokens. An alternative to this approach is to dynamically generate a new PKI certificate representing the user and allow the boundary device to mutually authenticate with the resource as that user. While this approach is often looked at with concern as the boundary device can impersonate the user for other actions, it can be thought of as an alternative authentication assertion, constrained and structured to be clear that it’s from the boundary device, acting as a proxy, asserting that the user truly did authenticate.

In this approach, the boundary device must be a trusted component of a secure communications path in what may be framed as a consensual “man-in-the-middle”, or possibly “men-in-the-middle” when implemented in a bilateral deployment. As such, this framework replaces the overall end-to-end mutual authentication with a segmented head-to-tail chain of mutually authenticated point-to-point connections through trusted boundary devices. Clearly for the overall connection to retain the same assurance level across the termination and regeneration of the boundary connection, the trusted boundary devices must operate and impersonate the initiating PKI credential with the same assurance associated with the initial issuance of the original user credential. 

It is appropriate to require specific security controls on boundary devices that impersonate or delegate through an ephemeral dynamic PKI credential issuance capability so that relying parties may be reasonably certain of the integrity of the original credential validation across the overall connection and remain protected against several known attack vectors.

**Comment by LarryFrank** [comment]
While I agree that "It is appropriate to require specific security controls on boundary devices that impersonate or delegate through an ephemeral dynamic PKI credential issuance capability so that relying parties may be reasonably certain of the integrity of the original credential validation across the overall connection and remain protected against several known attack vectors."  I think it is already covered.  This is a form of federation.  The "dynamically" created PKI certificates are simply an "assertion" that is presented in a way that allows the consuming to use it as it would a directly presented PKI certificate.  To be successful, the consuming entity needs to trust the PKI that issues it - in doing so, it needs to understand the security mechanisms associated with the issuance of those credentials including ID proofing and security of private keys - as it would for any PKI.  This is really no different than a SAML assertion - signed by the asserting authority and with lower overhead for a system that is already PK enabled.

**Comment by paul-grassi** [comment]
@jimfenton per our discussion with Tim, let's make sure that FIPS 140 requirements are applied at each FAL (and AAL).  Still waiting on response from Andy and Ray, but we know we need to fix that.

I will look at -C to see what the most appropriate language is, if any, to cover this type of assertion generation.
